src/librustdoc/html/render/mod.rs

```diff
@@ -2471,7 +2471,7 @@ fn item_function(w: &mut Buffer, cx: &Context<'_>, it: &clean::Item, f: &clean::
 fn render_implementor(
     cx: &Context<'_>,
     implementor: &Impl,
-    parent: &clean::Item,
+    trait_: &clean::Item,
```

Comment:
Are you sure this is the trait? I would expect it to be the impl block, in which case this seems fishy because `unstable` and `stable` are ignored on impls blocks.

Reply:
It is passed here:

https://github.com/rust-lang/rust/blob/4d0dd02ee07bddad9136f95c9f7846ebf3eb3fc5/src/librustdoc/html/render/mod.rs#L2803-L2805

where `it` comes from `item` in:

https://github.com/rust-lang/rust/blob/4d0dd02ee07bddad9136f95c9f7846ebf3eb3fc5/src/librustdoc/html/render/mod.rs#L1796

and `item` is the `Item` being matched on, so I assume it is the trait.

> I would expect it to be the impl block, in which case this seems fishy because `unstable` and `stable` are ignored on impls blocks.

Are they? They are used a lot in the standard library. Example: https://github.com/rust-lang/rust/blob/4d0dd02ee07bddad9136f95c9f7846ebf3eb3fc5/library/alloc/src/string.rs#L1821-L1826

Reply:
Nice archeology, you convinced me :)

> Are they? They are used a lot in the standard library. Example: 

Hmm, maybe just `unstable` is ignored. But it is definitely ignored: https://github.com/rust-lang/rust/blob/4d0dd02ee07bddad9136f95c9f7846ebf3eb3fc5/compiler/rustc_lint_defs/src/builtin.rs#L2810.

Reply:
Also I've tested setting the trait's `since` version to the same version as the impl, which worked (hid the version from the implementor section), so I think this is correct.